vendor/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto

```diff
@@ -1308,3 +1313,23 @@ message ReopenContainerLogRequest {
 
 message ReopenContainerLogResponse{
 }
+
+message CheckpointContainerRequest {
+    // ID of the container to be checkpointed
+    string container_id = 1;
```

Comment:
I chose to use a struct for the options of both the checkpoint and the restore request in [my extension](https://github.com/schrej/kubernetes/blob/feature/pod-migration/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto#L868-L902) so it's easier to extend in the future, for example if "actual" live-migration with pagefaultd, which would add a lot of parameters, should be supported in the future.
A struct makes it easier to add additional fields and also keeps the [interface used by kubernetes](https://github.com/schrej/kubernetes/blob/a9d7bcbe16427d26f81c0e9a7a193388ca9f6902/staging/src/k8s.io/cri-api/pkg/apis/services.go#L40-L43) clean. It's done the same way for other actions with lots of parameters.

Reply:
@schrej Thanks for your input. I opened a KEP at https://github.com/kubernetes/enhancements/pull/1990 about checkpoint/restore in the API. That is probably the place at which to have the API discussion. If structures are better suited then I am happy to use structures.